Add RRef.__str__() API

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30645 Apply clang-format to rpc/utils.cpp
* #30610 Adding Debug Info for RRef Context
* **#30609 Add RRef.__str__() API**



Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

mrshenli committed:
Add RRef.__str__() API

[ghstack-poisoned]

mrshenli committed:
Update on "Add RRef.__str__() API"


Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]

mrshenli committed:
Update on "Add RRef.__str__() API"


Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]

mrshenli committed:
Update on "Add RRef.__str__() API"


Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]

mrshenli committed:
Update on "Add RRef.__str__() API"




Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]

mrshenli committed:
Update on "Add RRef.__str__() API"




Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]

mrshenli committed:
Update on "Add RRef.__str__() API"




Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]

bddppq commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -144,7 +144,8 @@ PyObject* rpc_init(PyObject* /* unused */) {
                 return PyRRef::unpickle(t);
-              }));
+              }))
+          .def("__str__", [](const PyRRef& self) { return self.str(); });
```
I think you can do `.def("__str__", &PyRRef::str);` (without creating a lambda)

bddppq reviewed (approved):
LGTM

mrshenli committed:
Update on "Add RRef.__str__() API"




Differential Revision: [D18763593](https://our.internmc.facebook.com/intern/diff/D18763593)

[ghstack-poisoned]
